src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+
+namespace Microsoft.Graph.ODataTemplateWriter.PathWriters
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Graph.ODataTemplateWriter.Extensions;
+    using Microsoft.Graph.ODataTemplateWriter.Settings;
+    using Microsoft.Graph.ODataTemplateWriter.TemplateProcessor;
+
+    public class TypeScriptPathWriter : PathWriterBase
+    {
+
+        public override string WritePath(ITemplateInfo template, String baseFileName)
+        {
+            var theNamespace = this.CreateNamespace(template.OutputParentDirectory.ToLower());
+            var namespacePath = this.CreatePathFromNamespace(theNamespace);
+            var fileName = this.TransformFileName(template, baseFileName);
+            String filePath = Path.Combine(namespacePath, fileName);
+            return filePath;
+        }
+
+        private string CreateNamespace(string folderName)
+        {
+            var @namespace = this.Model.GetNamespace();
+            var prefix = ConfigurationService.Settings.NamespacePrefix;
+
+            if (String.IsNullOrEmpty(ConfigurationService.Settings.NamespaceOverride))
+            {
+                if (string.IsNullOrEmpty(folderName))
+                {
+                    return string.IsNullOrEmpty(prefix) ? @namespace
+                                                        : string.Format("{0}.{1}", prefix, @namespace);
+                }
+
+                return string.IsNullOrEmpty(prefix) ? string.Format("{0}.{1}", @namespace, folderName)
+                                                    : string.Format("{0}.{1}.{2}", prefix, @namespace, folderName);
+            }
+
+            @namespace = ConfigurationService.Settings.NamespaceOverride;
+            return folderName != null ? string.Format("{0}.{1}", @namespace, folderName)
+                                         : @namespace;
+        }
+
+        private string CreatePathFromNamespace(string @namespace)
+        {
+            var splittedPaths = @namespace.Split('.');
```

Comment:
This code was copy-pasted from another path writer. It looks like they all have this duplicate code unfortunately. I'd prefer for the TS file to not diverge from the others so when it's cleaned up we can easily find duplicates.